templates/eks/amazon-eks-nodepool-cf.yaml

```diff
@@ -654,14 +670,52 @@ Resources:
         SecurityGroupIds:
           !If [NoCustomNodeSecurityGroups, [!Ref NodeSecurityGroup], !Split [ ",",  !Join [ ",", [ !Ref NodeSecurityGroup, !Ref CustomNodeSecurityGroups ] ]  ] ]
         UserData: !Base64
-          "Fn::Sub": |
-            #!/bin/bash
-            set -o xtrace
-            /etc/eks/bootstrap.sh ${ClusterName} ${BootstrapArguments}
-            /opt/aws/bin/cfn-signal --exit-code $? \
-                     --stack  ${AWS::StackName} \
-                     --resource NodeGroup  \
-                     --region ${AWS::Region}
+            "Fn::Join":
```

Comment:
why do we use this format (individual !subs in a list of inline double quoted strings joined together) instead of the original multiline literal? https://yaml-multiline.info/

Or if this is a sample from aws (I couldn't find it in a similar state), please cite the original source.

Reply:
It's not a sample, I've put it in this form because Sub function appears to screw up things when there are bash env vars like here for ex. 'RAID_DEVICE=${RAID_DEVICE:-/dev/md0}' that's why I decided to use Sub fn only where it's necessary.

Reply:
I see. It seems much easier to read/maintain with the block literal and proper escaping, but handle this as an advice only.

> To write a dollar sign and curly braces (`${}`) literally, add an exclamation point (!) after the open curly brace, such as `${!Literal}`. AWS CloudFormation resolves this text as `${Literal}`.

Reply:
converted to block literal